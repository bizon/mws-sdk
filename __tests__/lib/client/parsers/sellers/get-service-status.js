const parseXml = require('../../../../../lib/client/parsers')
const parseGetServiceStatusResponse = require('../../../../../lib/client/parsers/base/get-service-status-response')

describe('lib.client.parsers.base.get-service-status-response', () => {
  it('should parse the GetServiceStatusResponse example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetServiceStatusResponse xmlns="https://mws.amazonservices.com/Sellers/2011-07-01">
        <GetServiceStatusResult>
          <Status>GREEN</Status>
          <Timestamp>2010-09-24T21:38:09.676Z</Timestamp>
        </GetServiceStatusResult>
      </GetServiceStatusResponse>`,
    )

    const response = parseGetServiceStatusResponse('/sellers:GetServiceStatusResponse', doc)

    expect(response).toMatchSnapshot()
  })
})
