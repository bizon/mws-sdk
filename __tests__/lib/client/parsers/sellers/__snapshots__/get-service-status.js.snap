// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.base.get-service-status-response should parse the GetServiceStatusResponse example response from MWS doc 1`] = `
{
  "getServiceStatusResult": {
    "status": "GREEN",
    "timestamp": 2010-09-24T21:38:09.676Z,
  },
  "responseMetadata": {
    "requestId": null,
  },
}
`;
